### PR TITLE
Updating prep and veteran selection for worksheet

### PR DESCRIPTION
**Pull request by tejans24**
connects #4037 #4113

### CODE REVIEW

File: client/app/hearings/DailyDocket.jsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
 import DocketHearingRow from './components/DocketHearingRow';
 import moment from 'moment';
 import { Link } from 'react-router-dom';
 import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
 
 export class DailyDocket extends React.Component {
 
+  saveDailyDocketInLocalStorage = (docket) => {
```
Comment (tejans24) [inline client/app/hearings/DailyDocket.jsx]: storing the daily docket information in local storage so that we don't have to make an additional request if we don't have to on the worksheet page.

File: client/app/hearings/util/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const getReaderLink = (worksheet) => `/reader/appeal/${worksheet.appeal_vacols_id}/documents`;
```
Comment (sharonwarner) [inline client/app/hearings/util/index.js]: I think this would be more reusable if we just sent the appeal_vacols_id rather than the worksheet.

File: client/app/hearings/reducers/index.js [hunk 3]
```diff
@@ -120,7 +140,10 @@ export const hearingsReducers = function(state = mapDataToInitialState(), action
     });
 
   case Constants.SET_HEARING_PREPPED:
-    return newHearingState(state, action, { prepped: { $set: action.payload.prepped } });
+    return setWorksheetPrepped(state, action, { prepped: { $set: action.payload.prepped } },
+      action.payload.setEdited);
+  case Constants.SET_WORKSHEET_HEARING_PREPPED:
```
Comment (sharonwarner) [inline client/app/hearings/reducers/index.js]: Can we use the same name for the action and the reducer? So SET_WORKSHEET_HEARING_PREPPED and setWorksheetHearingPrepped.

File: client/app/hearings/HearingWorksheet.jsx [hunk 4]
```diff
@@ -53,16 +56,15 @@ class WorksheetFormEntry extends React.PureComponent {
 }
 export class HearingWorksheet extends React.PureComponent {
 
-  componentDidMount() {
-    document.title = `${this.props.worksheet.veteran_fi_last_formatted}'s ${document.title}`;
+  componentDidUpdate(prevProps) {
+    if (prevProps.worksheet !== this.props.worksheet) {
+      document.title = `${this.props.worksheet.veteran_fi_last_formatted}'s ${document.title}`;
+    }
```
Comment (sharonwarner) [inline client/app/hearings/HearingWorksheet.jsx]: Would it work to move this to the PageRoute component in index.jsx? That would be a little cleaner.

File: client/app/hearings/actions/Dockets.js [hunk 5]
```diff
@@ -175,22 +186,88 @@ export const setWorksheetSaveFailedStatus = (saveFailed) => ({
   }
 });
 
-export const saveWorksheet = (worksheet) => (dispatch) => {
+export const saveWorksheet = () => (dispatch, getState) => {
+
+  const { worksheet } = getState();
```
Comment (sharonwarner) [inline client/app/hearings/actions/Dockets.js]: Hmm... I'm not sure if this makes sense to set worksheet like this.

File: client/app/hearings/actions/Dockets.js [hunk 6]
```diff
@@ -175,22 +186,88 @@ export const setWorksheetSaveFailedStatus = (saveFailed) => ({
...
+    moment(date).format('YYYY-MM-DD'), false));
+
+  ApiUtil.patch(`/hearings/${hearingId}`, { data: { prepped } }).
+    then((response) => {
+      dispatch(setHearingPrepped(hearingId, response.body.prepped,
+        moment(date).format('YYYY-MM-DD'), false));
+    },
+    () => {
+      // we need better error handling here
+      // eslint-disable-next-line no-console
```
Comment (sharonwarner) [inline client/app/hearings/actions/Dockets.js]: Yeah, generally our error handling isn't great in our save functions. We should update that in a bunch of places.

File: client/app/hearings/components/HearingWorksheetStream.jsx [hunk 7]
```diff
@@ -20,7 +20,7 @@ class HearingWorksheetStream extends Component {
         maxValue = Number(issue.vacols_sequence_id);
       }
     });
-
+    console.log(maxValue);
```
Comment (sharonwarner) [inline client/app/hearings/components/HearingWorksheetStream.jsx]: Was this just for debugging?

File: client/app/hearings/constants/constants.js [hunk 8]
```diff
@@ -16,7 +16,6 @@ export const SET_AOD = 'SET_AOD';
 export const SET_TRANSCRIPT_REQUESTED = 'SET_TRANSCRIPT_REQUESTED';
 export const SET_HEARING_VIEWED = 'SET_HEARING_VIEWED';
 export const SET_HEARING_PREPPED = 'SET_HEARING_PREPPED';
-
```
Comment (sharonwarner) [inline client/app/hearings/constants/constants.js]: Can we just recheckout this file so we don't unnecessary changes?

File: client/app/hearings/actions/Dockets.js (hunk 5 above)
Comment (tejans24) [reply client/app/hearings/actions/Dockets.js]: oops, good catch, this is from earlier. I'll remove this.

Comment (tejans24) [reply client/app/hearings/actions/Dockets.js]: this is just like being passed to this method, but it's rather getting it directly from the redux store.

File: client/app/hearings/HearingWorksheet.jsx (hunk 4 above)
Comment (tejans24) [reply client/app/hearings/HearingWorksheet.jsx]: the problem is that we need to change the title when the worksheet information changes. It doesn't make much sense for the PageRoute to know about the worksheet object.

File: client/app/hearings/actions/Dockets.js (hunk 5 above)
Comment (tejans24) [reply client/app/hearings/actions/Dockets.js]: But I will make the change, since it might be a slightly confusing.

File: client/app/hearings/components/HearingWorksheetStream.jsx (hunk 7 above)
Comment (tejans24) [reply client/app/hearings/components/HearingWorksheetStream.jsx]: yep, oops
